app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
+  respond_to :js, :html
+
+  def index
+    @types = current_user.types.order('created_at DESC').page(params[:page]).per(4)
+
+    @types = if params[:show_all] == 'true'
+               Type.all.order('created_at DESC').page(params[:page]).per(4)
+             else
+               current_user.types.order('created_at DESC').page(params[:page]).per(4)
+             end
```

Comment:
@basti the usage of lines from 10 to 14 is because of page index.html.erb, I have two buttons, the first is to display all created types, and the second is to display only types that the user created. I transfer value with the URL and check which button was clicked. Line 8 displays the default value of @types, which shows each time on /types URL. Should I change logic, besides if statement?

Reply:
@basti I also have a problem, with the ternary operator, because it doesn't recognize params[:show_all] it is true or false, because it doesn't check boolean value, it checks if the path of URL is equal to true or false, so it won't work through the ternary operator.

I could implement that functionality by creating an additional method and route for that method, but I am not sure if that is RESTful.

I also have a question why do you not recommend that this be done in an already implemented way?

Reply:
@VanjaJeftic if lines 10 through 14 is because of index.html.erb, what is line 8 used then for? Try just deleting line 8 and the app should continue to work exactly the same. That was my point. I didn’t say you need to change the implementation.

The only implementation comment was related to using `if` assignment which I personally dislike.

Oh, and ternary operator can have complex conditions (not that using it here would be the best choice because it would be long line), for example: `params[:show_all] == true ? something : something_else`.